packages/shopping/src/services/user-service.ts

```diff
@@ -0,0 +1,52 @@
+// Copyright IBM Corp. 2019. All Rights Reserved.
+// Node module: @loopback/authentication
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+import {HttpErrors} from '@loopback/rest';
+import {Credentials, UserRepository} from '../repositories/user.repository';
+import {User} from '../models/user.model';
+import {UserService, UserProfile} from '@loopback/authentication';
+import {repository} from '@loopback/repository';
+import {PasswordHasher} from './hash.password.bcryptjs';
+import {PasswordHasherBindings} from '../keys';
+import {inject} from '@loopback/context';
+
+export class MyUserService implements UserService<User, Credentials> {
+  constructor(
+    @repository(UserRepository) public userRepository: UserRepository,
+    @inject(PasswordHasherBindings.PASSWORD_HASHER)
+    public passwordHasher: PasswordHasher,
+  ) {}
+
+  async verifyCredentials(credentials: Credentials): Promise<User> {
+    const foundUser = await this.userRepository.findOne({
+      where: {email: credentials.email},
+    });
+
+    if (!foundUser) {
+      throw new HttpErrors.NotFound(
+        `User with email ${credentials.email} not found.`,
+      );
+    }
+    const passwordMatched = await this.passwordHasher.comparePassword(
+      credentials.password,
+      foundUser.password,
+    );
+
+    if (!passwordMatched) {
+      throw new HttpErrors.Unauthorized('The credentials are not correct.');
+    }
+
+    return foundUser;
+  }
+
+  convertToUserProfile(user: User): UserProfile {
+    // since first name and surname are optional, no error is thrown if not provided
+    let userName = '';
+    if (user.firstname && user.surname) {
```

Comment:
Question: are they allowed to provide only `firstname` or `surname` or does it have to be both to be used as the `name`?

Reply:
@nabdelgadir . Just a personal decision at the time of writing. But you raise a good point.

Reply:
I would prefer `firstName` and `lastName`.

Reply:
@raymondfeng , I have addressed this in my latest commit